sympy/physics/units/util.py

```diff
@@ -148,6 +149,8 @@ def quantity_simplify(expr):
                 coeff = coeff * arg.scale_factor
         expr = coeff
 
+    if any(isinstance(a, Add) for a in expr.args):
```

Comment:
You have already run quantity_simplify over all arguments at line 138. I don't think these added lines will ever do anything.

Reply:
Oh...you are trying to avoid the error you identified in the OP. This will avoid the error but then will fail to simplify something like:

```
>>> quantity_simplify(foot*inch*(foot + inch))
foot*inch*(foot + inch)
```